### PR TITLE
docs: add contribution guidelines

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
We encourage contributions from all SKY UX users. To contribute, follow this step-by-step guide. For more additional guidance, see the [SKY UX Contribution Guidelines](https://developer.blackbaud.com/skyux/contribute/contribution-process/guidelines).
```
Disregard if you don't like these edits. I thought we should have "the" before the link in the last sentence and tweaked the opening clause to make it flow slightly better.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
- [Contribution process](#contribution-process)
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
## <a name="contribution-process"></a> Contribution process
```
Potentially dumb question: In these H2s, should the header text be _inside_ the `a` tags instead of after the closing tag?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
- [Coding rules](#coding-rules)
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
## <a name="coding-rules"></a> Coding rules
```
Do we need to add this to the docs on the website?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
## <a name="commit-message-guidelines"></a> Commit message guidelines
```
Do we need to add this to the docs on the website?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
- [Commit message guidelines](#commit-message-guidelines)
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
- [Local development cheat sheet](#local-development)
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
## <a name="local-development"></a> Local development cheat sheet
```
Do we need to add this to the docs on the website?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]: On the docs site we say "and mention the `@skyux-team`" in one place and "tag the SKY UX team" in a second place. ... Should those be updated to "mention the `@blackbaud/skyux-maintainers` team" like we have here?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]: On the docs site, do we need to change `@skyux-team` to `@blackbaud/skyux-maintainers`?
```suggestion
After you file an issue, discuss the architecture and implementation details, requirements, and timelines with the `@blackbaud/skyux-maintainers` team. We will assign a developer to your issue to help answer questions.
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. Create a branch off of `main`. Name it after the feature to contribute, and use kebab-case (`my-new-feature`).
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. Clone your repo locally and run `nvm use` to ensure you have the preferred version of Node.js installed. Then run `npm install` to install all required dependencies. For more information on prerequisites, see [SKY UX prerequisites](https://developer.blackbaud.com/skyux/learn/get-started).
```
When we flip the switch to make v5 the primary site, the prerequisites will be directly on the getting-started page, not on a child page under it.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md:28]: Do we need a "Write the code" step here like we have on the docs site?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md:37]: Is the part about visual regression tests correct? We removed that section of the contribution guidelines from the v5 docs as part of Braden's review and cleanup of the contribution guidelines, and I just realized that we still have this broken link in the contribution process tutorial. If visual regression tests are still necessary, then our v5 docs are wrong now and we need to add that content back to the contribution guidelines or that link will not work after we flip the switch to make v5 the primary site. If they are not required, we need to remove the link here, and I need to remove the soon-to-be-busted link from the contribution process tutorial.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. [Squash all commits on your feature branch into one](#squashing-all-commits-into-one). This consolidates all your work and includes a well-formed commit message to use in the changelog.
```
Do we need to call this part of the final review step out in the guidance on the docs site?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. Mention the `@blackbaud/skyux-maintainers` team to review the complete contribution. We review the implementation, testing, functionality, styling, localization, and accessibility.
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
#### Address review feedback
```
Do we need to add this to the docs on the website?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
#### Update the commit message
```
Do we need to add this to the docs on the website?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
If the reviewer asks you to reword your commit message, use the `--amend` flag:
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
_Note: To amend an previous commit, you need to [squash all commits into one](#squashing-all-commits-into-one)._
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
#### Squash all commits into one
```
Do we need to add this to the docs on the website?

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
If your pull request branch has multiple non-fixup commits, you need to squash all commits into one commit with a message that meets the [`conventional-changelog` specification](https://github.com/conventional-changelog/conventional-changelog).
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
Run the following command to squash all commits in your feature branch with a new commit message that you provide. Local changes remain intact, but this rewrites your feature branch's commit history.
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. All features and bug fixes must include at least one test spec and must not lower the overall code coverage percentage.
```
I assume this is a requirement rather than a suggestion, but if it's optional, I'd stick with `should`.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. Provide JSDocs comments for all public API component inputs and outputs and all public service methods.
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. Style code using our Prettier rules.
```

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
- See the [full list of supported **scopes**](https://github.com/blackbaud/skyux/blob/main/.github/workflows/validate-pr.yml).
```

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: Definitely a mistake. Good catch!

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md:37]: Good question. We're working on re-adding visual tests to our monorepo, so we can either keep that section in the docs, or remove it now (and re-add it later).

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: No, this is specific to the monorepo.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: No, this is specific to the monorepo.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: No, this is specific to the monorepo.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: No, this is specific to the monorepo.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: No, this is specific to the monorepo.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: No, this is specific to the monorepo.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: I just did some testing on this, and only members within the Github `blackbaud` organization can "@" mention team names. Outside contributors may only "@" mention specific user names.

When we say "mention the @skyux-team", that's specific to our Slack team (the equivalent team name on Github is "@blackbaud/skyux-maintainers") so I'm guessing we don't want to include "@skyux-team" on the public docs. Might be good to put that under a restricted-view component.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md]: No, this is only related to monorepo.

File: CONTRIBUTING.md
Comment (Blackbaud-SteveBrush) [reply CONTRIBUTING.md:28]: I didn't think it was necessary due to it being implied by the first sentence of the "Request a preliminary review"?

Comment (blackbaud-johnly) [reply CONTRIBUTING.md:28]: If you don't think we need a section for that step in the CONTRIBUTING.md file, I think that's fine. I just wasn't sure since we copy most of the stuff from the docs site and we include it as a separate step there.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [reply CONTRIBUTING.md]: I'll set up a PR to rework those references so that they work in the public docs.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [reply CONTRIBUTING.md:37]: We already removed the section, so I'll remove the link to it for now, and we can re-add it when necessary. Thanks.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
_Note: To amend a previous commit, you need to [squash all commits into one](#squash-all-commits-into-one)._
```
I assume we need to tweak this link to reflect the change to the header. Is that right? (Of course, given that it's linking to the next line in the file, maybe the link isn't really necessary.)

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
We encourage contributions from all SKY UX users. To contribute, follow this step-by-step guide. For additional guidance, see the [SKY UX Contribution Guidelines](https://developer.blackbaud.com/skyux/contribute/contribution-process/guidelines).
```
I missed deleting "more" in my original suggestion here. Sorry.

File: CONTRIBUTING.md
Comment (blackbaud-johnly) [inline CONTRIBUTING.md]:
```suggestion
1. Before requesting a final review, [squash all commits on your feature branch into one](#squash-all-commits-into-one). This will consolidate all your work and include a well-formed commit message to be used in the changelog.
```

**Review by blackbaud-johnly — APPROVED** [review]

**Pull request merged**
